pkg/volume/csi/csi_mounter.go

```diff
@@ -254,10 +257,13 @@ func (c *csiMountMgr) SetUpAt(dir string, mounterArgs volume.MounterArgs) error
 	)
 
 	if err != nil {
-		if removeMountDirErr := removeMountDir(c.plugin, dir); removeMountDirErr != nil {
-			klog.Error(log("mounter.SetupAt failed to remove mount dir after a NodePublish() error [%s]: %v", dir, removeMountDirErr))
+		// If operation finished with error then we can remove the mount directory.
```

Comment:
MkdirAll() can be called right before NodePublishVolume call, right? so all previous errors does not need to worry about remove dir.